packages/fxa-payments-server/src/routes/Product/index.tsx

```diff
@@ -118,23 +118,27 @@ export type ProductProps = {
   profile: SelectorReturns['profile'];
   plans: SelectorReturns['plans'];
   customer: SelectorReturns['customer'];
+  subscriptionChangeEligibility: SelectorReturns['subscriptionChangeEligibility'];
```

Comment:
As you might know, in the near future we'd like to remove Redux from the `payments-server`.

Therefore, I don't think we should be adding on to our Redux code, unless there's a specific reason?

Reply:
I had initially added, and then removed it after discussing with Ivo on Monday, November 7, 2022. But then in our meeting on November 9, 2022, where Ivo asked the team about it, the answer was that while it will eventually be removed, it wasn't being removed yet, and it hadn't been decided on exactly when (but almost certainly in the new year).

After careful consideration, I chose to add it this way for a few reasons:

1. Consistency with how loading is handled right now (at least in `routes/Product`). When it gets updated, it can be updated together or done it sequence (that is, it makes the refactor process more seamless with less of a chance something gets missed or has to be updated again). (Same applies to tests.)
2. Plan eligibility improvements are outlined in FXA-6166 that might affect how loading is handled (it might not, as I don't know how that ticket will be handled yet; but this reduces the potential for extra work by keeping it as it is currently (see 1)).
3. Ticket scope was evaluated based on how it was done before and how much time and effort would be needed to add it in, as an addition to FXA-5080 that didn't initially account for it.

Reply:
OK that makes sense.

Unfortunately, from the looks of it, at the moment half the logic is in Redux and half of it is in the Hook. Looks like the plan eligibility result is read from the direct API call, where as the loading state is read from the Redux store.

Also at the moment it looks like `/plan-eligibility` is called twice. Once from a direct call to the API and once via the Redux Thunk. 

I think it's fine to use either Redux or Hooks, but we shouldn't have both.

I'd advocate for using Hooks, instead of Redux, since then we could in future create a custom Hook for this logic, and more easily use it on other pages if necessary.

Reply:
Ah good catch! I completely forgot to update that. I was so focused on loading. Thank you! 👍

Totally aligned with everything and for the very reasons you mentioned! Definitely the direction we should head. For now, I went the other direction solely because of the aforementioned reasons. Please let me know though if I missed anything else 🙂.